src/main/java/com/puppycrawl/tools/checkstyle/AstTreeStringPrinter.java

```diff
@@ -41,6 +41,20 @@
  */
 public final class AstTreeStringPrinter {
 
+    /**
+     * Enum to be used for test if comments should be printed.
+     */
+    public enum PrintOptions {
+        /**
+         * Comments has to be printed.
+         */
+        COMMENTS,
+        /**
+         * Comments has NOT be printed.
+         */
+        NONE
+    }
```

Comment:
@Nimfadora ,

you can ignore comment of rnveach, we had long conversation in private on this and he agree with me to some extend.

Notes:
nuance is that I do not plan to extend this Enum
if that is going to grow options as set of independent options .... not a state/type (Enum value is whole state or type)
options is collection of some values {WITH_COMMENTS, NO_JAVADOC, HIGHLIGHT_SYNTAX} - so it will result in usage of BitSet type.
I think that BitSet type usage is too much for this simple internal item.
When we come to more options for printer and Enum will not satisfy us, we will do refactoring to better model.